src/representation_theory/fdRep.lean

```diff
@@ -115,26 +129,26 @@ namespace fdRep
 
 open representation
 
-variables {k G V W : Type u} [field k] [group G]
-variables [add_comm_group V] [module k V] [add_comm_group W] [module k W]
-variables [finite_dimensional k V] [finite_dimensional k W]
-variables (ρV : representation k G V) (ρW : representation k G W)
+variables {k G V : Type u} [field k] [group G]
+variables [add_comm_group V] [module k V]
+variables [finite_dimensional k V]
+variables (ρV : representation k G V) (W : fdRep k G)
```

Comment:
This is a bit too weird for me, living half in `representation` and half in `fdRep`. What's going on?

Reply:
Ideally when we replace `lin_hom` and `dual` by monoidal closed and rigid structures both variables would live in `fdRep`, but for now we need to use `representation`. I could alternatively take `(V : fdRep)`, but then the statement of `dual_tensor_iso_lin_hom` would be
```
def dual_tensor_iso_lin_hom : (fdRep.of V.ρ.dual) ⊗ W ≅ fdRep.of (lin_hom V.ρ W.ρ)
```
which follows directly from the current version applied to `V.ρ`. I chose the current version since it's strictly more general, as it applies to a representation `ρV` that's not necessarily of the form `V.ρ`, but it does have the disadvantage of having these slightly awkward variables.

Reply:
I guess I don't really mind here, perhaps a slight preference for just having both `V W : fdRep k G`.

However I think this PR will really need a comment explicitly explaining that the situation is currently a bit broken, and that we would like to in future use the monoidal closed and rigid structures.

Reply:
I already mention that in the TODO section of `representation_theory/character`, but I've also added a comment in this section.